File: cpp/etl/etl.cpp
#include "etl.h"
#include <cctype>

using namespace std;

map<char, int> etl::transform(const map<int, vector<char>>& old) {
    map<char, int> out;

    for (const auto& p : old) {
        auto val = p.first;
        auto vec = p.second;

        for (const auto& c : vec) {
            out[tolower(c)] = val;
        }
    }

    return out;
}

File: csharp/etl/Etl.cs
using System;
using System.Collections.Generic;

public static class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old)
    {
        var output = new Dictionary<string, int>();

        foreach (var entry in old) {
            foreach(var val in entry.Value)
                output[val.ToLower()] = entry.Key;
        }

        return output;
    }
}
